crates/downloader/src/bundle_buffer.rs

```diff
@@ -0,0 +1,172 @@
+use std::collections::HashMap;
+
+use fuel_block_committer_encoding::{
+    blob::{
+        self,
+        Blob,
+        Header,
+        HeaderV1,
+    },
+    bundle::{
+        self,
+        Bundle,
+        BundleV1,
+    },
+};
+
+/// Holds a buffer for each incomplete bundle. Bundles are removed from the buffer when they are complete.
+#[derive(Default)]
+pub struct IncompleteBundleBuffers {
+    /// Mapping from bundle id to a sorted list of blobs
+    bundle_buffer_by_id: HashMap<u32, Vec<(HeaderV1, Blob)>>,
```

Comment:
```suggestion
    bundle_buffer_by_id: HashMap<u32, Vec<(Header, Blob)>>,
```
can we reuse Header here instead of HeaderV1?

Reply:
addressed in [7c38730](https://github.com/FuelLabs/network-watchtower/pull/10/commits/7c38730ff385559317b3e75499a7195d5cf710aa)